homeassistant/components/media_player/directv.py

```diff
@@ -64,29 +65,59 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         name = 'DirecTV_{}'.format(discovery_info.get('serial', ''))
 
         # Attempt to discover additional RVU units
+        _LOGGER.debug("Doing discovery of DirecTV devices on %s", host)
         try:
             resp = requests.get(
                 'http://%s:%d/info/getLocations' % (host, DEFAULT_PORT)).json()
-            if "locations" in resp:
-                for loc in resp["locations"]:
-                    if("locationName" in loc and "clientAddr" in loc
-                       and loc["clientAddr"] not in known_devices):
-                        hosts.append([str.title(loc["locationName"]), host,
-                                      DEFAULT_PORT, loc["clientAddr"]])
+
+            _LOGGER.debug("Known devices: %s", known_devices)
+            for loc in resp.get("locations") or []:
```

Comment:
@MartinHjelmare , 
Using getLocations() instead to retrieve the locations. Going through the data received is  done to confirm that the device has not already been created as an entity in HASS.

Reply:
That's better. But we should not have to parse the json response and know the correct strings to look for to use the library. The library should implement an object model where we access the attributes of the objects. The objects and attributes of the objects should represent the features of the device.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services

I think it could be ok to merge this for now, but in my opinion the library should take more responsibility before we can take any future changes to this integration.